rust/kernel/chrdev.rs

```diff
@@ -121,8 +125,10 @@ impl<const N: usize> Registration<{ N }> {
             (*cdev).owner = this.this_module.0;
             let rc = bindings::cdev_add(cdev, inner.dev + inner.used as bindings::dev_t, 1);
             if rc != 0 {
+                bindings::cdev_del(cdev);
```

Comment:
I'm starting to think that we need a wrapper for a `cdev`,  such that clean up is automatic. Without it, it is very much possible for us to add an extra error path and forget to clean up; this is the sort of thing we're trying to be better at than C.

Reply:
I was thinking the exact same thing while I was typing up the `cdev_del` in the error path. That didn't feel very oxidated!

A wrapper/abstraction would be cool. The more `kobject` instances we need to juggle around in Rust, the more useful such an abstraction would be. I guess its complexity has to balance its benefit?